azkaban-common/src/main/java/azkaban/executor/AlerterHolder.java

```diff
@@ -16,119 +16,118 @@
 package azkaban.executor;
 
 import azkaban.alert.Alerter;
+import azkaban.spi.AzkabanException;
 import azkaban.utils.Emailer;
 import azkaban.utils.FileIOUtils;
 import azkaban.utils.PluginUtils;
 import azkaban.utils.Props;
 import azkaban.utils.PropsUtils;
-import javax.inject.Inject;
-import javax.inject.Singleton;
 import java.io.File;
 import java.lang.reflect.Constructor;
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import org.apache.log4j.Logger;
-
+import java.util.function.BiConsumer;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Singleton
 public class AlerterHolder {
 
-  private static final Logger logger = Logger.getLogger(AlerterHolder.class);
-  private Map<String, Alerter> alerters;
+  private static final Logger logger = LoggerFactory.getLogger(AlerterHolder.class);
+  private final Map<String, Alerter> alerters;
 
   @Inject
   public AlerterHolder(final Props props, final Emailer mailAlerter) {
-    try {
-      this.alerters = loadAlerters(props, mailAlerter);
-    } catch (final Exception ex) {
-      logger.error(ex);
-      this.alerters = new HashMap<>();
-    }
-  }
-
-  private Map<String, Alerter> loadAlerters(final Props props, final Emailer mailAlerter) {
     final Map<String, Alerter> allAlerters = new HashMap<>();
-    // load built-in alerters
+    // Load built-in alerters.
     allAlerters.put("email", mailAlerter);
-    // load all plugin alerters
-    final String pluginDir = props.getString("alerter.plugin.dir", "plugins/alerter");
-    allAlerters.putAll(loadPluginAlerters(pluginDir));
-    return allAlerters;
+    // Load additional alerter plugins if any.
+    final String customAlertersBaseDir = props.getString("alerter.plugin.dir", "plugins/alerter");
+    allAlerters.putAll(loadCustomAlerters(customAlertersBaseDir));
+    this.alerters = allAlerters;
+    logger.info("Alerter plugins loaded: {}", this.alerters.keySet());
   }
 
-  private Map<String, Alerter> loadPluginAlerters(final String pluginPath) {
-    final File alerterPluginPath = new File(pluginPath);
-    if (!alerterPluginPath.exists()) {
+  private Map<String, Alerter> loadCustomAlerters(final String alerterPluginsRootPath) {
+    final File pluginsRootPath = new File(alerterPluginsRootPath);
+    if (!pluginsRootPath.exists()) {
```

Comment:
minor: maybe add log here

Reply:
I think we can use log in line 52 to learn if custom alerters are configured or not. As they might not be configured too often adding a log here could be just noise most of the times.